hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/records/ContractUpdateRecordBuilder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.records;
+
+import com.hedera.hapi.node.base.ContractID;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Exposes the record customizations needed for a HAPI contract update transaction.
+ */
+public interface ContractUpdateRecordBuilder {
+
+    /**
+     * Tracks the contract id created by a successful top-level contract creation.
+     *
+     * @param contractId the {@link ContractID} of the new top-level contract
+     * @return this builder
+     */
+    @NonNull
+    ContractUpdateRecordBuilder contractID(@Nullable ContractID contractId);
```

Comment:
Probably we don't need this, since we are not updating record for ContractUpdate.